### ISSUE TITLE
`serialize(options("askpass"))` and other RStudio options generates unwanted warning

### ISSUE THREAD

**Issue by mtmorgan** [rstudio/rstudio#3946]
### System details

    RStudio Edition : Desktop
    RStudio Version :  1.2.1139
    OS Version      :  macOS
    R Version       :  R Under development (unstable) (2018-11-25 r75670)

### Steps to reproduce the problem

```
xx <- serialize(options("askpass"), NULL)
```

### Describe the problem in detail

Only RStudio options generate the warning

```
> xx = serialize(options("askpass"), NULL)
Warning message:
In serialize(options("askpass"), NULL) :
  'package:stats' may not be available when loading
```

The context is at https://support.bioconductor.org/p/115415/ with a simpler reproducible example

```
install.package("BiocManager")
BiocManager::install("BiocParallel")
library(BiocParallel)
bplapply(1:2, function(...) {})
```

The problematic options are found with

```
bar <- function() {
  opt = options()
  nms = names(Filter(Negate(is.null), lapply(opt, environment)))
  bad <- NULL
  for (nm in nms) {
    withCallingHandlers({
      serialize(opt[[nm]], NULL)
    }, warning = function(x) {
      bad <<- c(bad, nm)
      invokeRestart("muffleWarning")
    })
  }
  bad
}
```
resulting in
```
> bar()
 [1] "askpass"                    "asksecret"                 
 [3] "buildtools.check"           "buildtools.with"           
 [5] "pager"                      "plumber.swagger.url"       
 [7] "profvis.print"              "restart"                   
 [9] "reticulate.repl.hook"       "reticulate.repl.initialize"
[11] "reticulate.repl.teardown"   "shiny.launch.browser"      
```

`error` and `terminal.manager` also seem to be problematic.

### Describe the behavior you expected

One does not expect a warning, as with all the other global options.

**Comment by kevinushey** [comment]
This occurs for each option we set with a direct reference to an internal RStudio function; e.g. for `askpass` we have:

https://github.com/rstudio/rstudio/blob/c407111662ab93b70c0854f285a1a5bf3ecec6b3/src/cpp/r/R/Options.R#L90-L91

I believe using a wrapper function would help here, e.g.

    options(askpass = function(...) .rs.askForPassword(...))

**Comment by kevinushey** [comment]
I think this is in general a problem for any function defined within an environment on the search path, e.g. RStudio does something like this for putting its own tools on the search path:

``` r
envir <- attach(NULL, name = "tools:rstudio")
f <- function() {}
environment(f) <- envir
serialize(f, NULL)  # warns
```

And this normally puts `tools:rstudio` in front of `package:stats`, e.g.

``` r
> search()
 [1] ".GlobalEnv"        "tools:rstudio"     "package:stats" <...>
```

But this error is always going to occur as long as these functions are defined within an environment on the search path (unless of course they're just in front of base or Autoloads).

**Comment by mtmorgan** [comment]
...or in a package?

**Comment by kevinushey** [comment]
We did previously try placing some functions in an auto-installed R package but this caused a litany of problems (since now RStudio must get into the business of auto-installing a package on the user's system and this can and did fail in a myriad of ways)

**Comment by kevinushey** [comment]
Tracking in https://github.com/rstudio/rstudio/issues/7001.

**Comment by dfalty** [comment]
Verified fixed using steps given on version 1.4.874.
